### DISCUSSION TITLE
Question About the Implementation in monotonicity_preserving_solver.h

### DISCUSSION THREAD

**Discussion by SADPR** [KratosMultiphysics/Kratos#11573] [Q&A] [answered]
Hi @ddiezrod (Daniel Diez),

I've been delving into the `monotonicity_preserving_solver.h` within the Kratos repository and I'm eager to learn more about its origins. Could you possibly guide me to any papers or references that detail or influenced this implementation?

P.S. If anyone else in the Kratos community has information or references related to this, I'd greatly appreciate your input as well.

Thanks in advance for any assistance!

**Comment by RiccardoRossi** [comment, marked as answer]
take a look at the book by Dimitri Kuzmin. @rubenzorrilla can pass it to you
